scylla/src/transport/errors.rs

```diff
@@ -261,6 +261,10 @@ pub enum ProtocolError {
     #[error("Schema version fetch protocol error: {0}")]
     SchemaVersionFetch(SingleRowTypedError),
 
+    /// A protocol error appeared during cluster metadata fetch.
+    #[error("Cluster metadata fetch protocol error: {0}")]
+    Metadata(#[from] MetadataProtocolError),
+
```

Comment:
Since I think it makes sense to do it for this variant. However, IMO it's not true for variants such as `UnexpectedResponse(CqlResponseKind)` - here, `CqlResponseKind` is not an error type itself, it's there only to provide more context. The error itself is that we received an unexpected response - I don't think `From<CqlResponseKind>` impl makes sense here.